src/snowflake/cli/plugins/git/commands.py

```diff
@@ -64,3 +69,69 @@ def fetch(
     **options,
 ) -> CommandResult:
     return QueryResult(GitManager().fetch(repo_name=repository_name))
+
+
+@app.command(
+    "copy",
+    help="Copies all files from given state of repository to local directory or stage.",
+    requires_connection=True,
+)
+def copy(
+    repository_path: str = RepoPathArgument,
+    destination_path: str = typer.Argument(
+        help="Target path for copy operation. Should be stage or local file path.",
+    ),
+    parallel: int = typer.Option(
+        4,
+        help="Number of parallel threads to use when downloading files.",
+    ),
+    **options,
+):
+    _assert_repository_path_is_stage("REPOSITORY_PATH", path=repository_path)
```

Comment:
Implement this as callback on `RepoPathArgument`

Reply:
Adressed in https://github.com/snowflakedb/snowflake-cli/pull/843